yarn-project/pxe/src/storage/note_data_provider/note_data_provider.test.ts

```diff
@@ -1,469 +1,581 @@
-    noteDataProvider = await NoteDataProvider.create(store);
+      const notesA = await provider2.getNotes({ contractAddress: contractA });
+      const notesB = await provider2.getNotes({ contractAddress: contractB });
+
+      expect(notesA.map(n => n.index)).toEqual([BigInt(1)]);
+      expect(notesB.map(n => n.index)).toEqual([BigInt(2)]);
+
+      await store.close();
+    });
   });
 
-  /**
-   * Sets up test data before each test case.
-   *
-   * This creates a predictable dataset of 10 notes distributed across:
-   * - 2 recipients (users who can receive notes)
-   * - 2 contract addresses (smart contracts that manage notes)
-   * - 2 storage slots (locations within contracts where notes are stored)
-   *
-   * The notes are distributed in a round-robin fashion using modulo arithmetic,
-   * ensuring each combination of (contract, slot, recipient) has predictable coverage.
-   *
-   * Data Distribution Pattern:
-   * - Notes 0,2,4,6,8: contract[0], slot[0], recipient[0], blocks 0,2,4,6,8
-   * - Notes 1,3,5,7,9: contract[1], slot[1], recipient[1], blocks 1,3,5,7,9
-   *
-   * This setup enables comprehensive testing of filtering combinations:
-   * - Single dimension filters (by contract, slot, or recipient)
-   * - Multi-dimension filters (e.g., contract AND slot)
-   * - Edge cases (non-existent values should return empty results)
-   */
-  beforeEach(async () => {
-    recipients = await timesParallel(2, () => AztecAddress.random());
-    contractAddresses = await timesParallel(2, () => AztecAddress.random());
-    storageSlots = times(2, () => Fr.random());
-
-    // Create 10 notes with deterministic distribution across the above entities
-    // Uses modulo (%) to cycle through arrays, ensuring even distribution
-    notes = await timesParallel(10, i => {
-      return NoteDao.random({
-        contractAddress: contractAddresses[i % contractAddresses.length], // Alternates between 2 contracts
-        storageSlot: storageSlots[i % storageSlots.length], // Alternates between 2 slots
-        recipient: recipients[i % recipients.length], // Alternates between 2 recipients
-        index: BigInt(i), // Sequential index: 0,1,2,...,9
-        l2BlockNumber: i, // Sequential block: 0,1,2,...,9
+  describe('NoteDataProvider.getNotes filtering happy path', () => {
+    let store: AztecLMDBStoreV2;
+    let provider: NoteDataProvider;
+    let note1: NoteDao; // eslint-disable-line @typescript-eslint/no-unused-vars
+    let note2: NoteDao;
+    let note3: NoteDao;
+
+    beforeEach(async () => {
+      ({ store, provider, note1, note2, note3 } = await setupProviderWithNotes('note_data_provider_get_notes_happy'));
+    });
+
+    afterEach(async () => {
+      await store.close();
+    });
+
+    it('filters notes matching only the contractAddress', async () => {
+      const res = await provider.getNotes({ contractAddress: contractA });
+      expect(res.map(n => n.index.toString()).sort()).toEqual(['1', '2']);
```

Comment:
```suggestion
      expect(res.map(n => n.index).sort()).toEqual([1n, 2n]);
```

The conversion to `string` is unnecessary here.

Reply:
Thanks - so I removed the `toString()` conversion. Sorting didn’t behave as I expected, which broke my tests because they were comparing strings '1', '2' rather than the BigNums 1n, 2n....  Which made me reconsider what the expect's are actually checking. 

- The DB returns notes in arbitrary order.
- Sorting BigInts without a comparator gave this:
```
const indexes = [1n, 10n, 2n, 4n];
indexes.sort(); 
// expected: [1n, 2n, 4n, 10n]
// got:      [1n, 10n, 2n, 4n]
```

I figured it’s better not to rewrite the expected results.   And adding a comparator for BigInt would just add more chaff.

So I updated the tests to use `Set` for equality checks and `.toContain` / `.not.toContain` where appropriate. This also ensures we’re checking _exactly_ the contents, not just “does it contain this… might also contain more.”